src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -151,6 +151,58 @@ void CascadiaSettings::_CreateDefaultProfiles()
     _profiles.emplace_back(powershellProfile);
     _profiles.emplace_back(cmdProfile);
     
+    // Add all WSL distros from: HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+    Profile WSLDistro{};
+    wil::unique_hkey hLXSSKey;
+    wil::unique_hkey hHKCUUncached;
+    RegOpenCurrentUser(KEY_READ, &hHKCUUncached); // Retrieve the uncached equivalent of HKEY_CURRENT_USER
+    if (RegOpenKeyExW(hHKCUUncached.get(), L"Software\\Microsoft\\Windows\\CurrentVersion\\Lxss",
+            0, KEY_READ, &hLXSSKey) == ERROR_SUCCESS)
+    {
+        std::filesystem::path AppsDirectory{ ExpandEnvironmentVariableString(L"%ProgramFiles%") };
+        AppsDirectory /= "WindowsApps";
+        std::filesystem::path WSLCommandLine{ L"" };
+
+        wchar_t WSLKey[MAX_PATH]{ L"" }; // Registry key for a distinct WSL Distribution
+        wchar_t DistributionName[MAX_PATH]{ L"" }; // The name to be displayed by the flyout list
+        wchar_t PackageFamilyName[MAX_PATH]{ L"" }; // The distinct name for the distribution
+        std::wstring PFNString{ L"" };
+        std::wstring PFNSubString{ L"" };
+
+        // These tell the Registry functions how much space they have to work with when returning a value.
+        DWORD cchWSLKey{ (DWORD) std::size(WSLKey) };
+        DWORD cchDistributionName{ (DWORD) std::size(DistributionName) };
+        DWORD cchPackageFamilyName{ (DWORD) std::size(PackageFamilyName) };
+
+        for (int i = 0;
+            RegEnumKeyExW(hLXSSKey.get(), i, WSLKey, &cchWSLKey, nullptr, nullptr, nullptr, nullptr) == ERROR_SUCCESS;
+            i++)
+        {
+            // Reset available space for each value.
+            cchWSLKey = (DWORD) std::size(WSLKey);

```

Comment:
Try to use the least powerful cast possible to effect the change needed, so if possible, use C++-style casts here.  For example, `static_cast<DWORD>(std::size(WSLKey))`.  This should be changed here and in any new code added.

[ES.49: If you must use a cast, use a named cast](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es49-if-you-must-use-a-cast-use-a-named-cast)